'cosign -version' tells me to use 'cosign --version'

cosign version 1.8.0 outputs a misleading warning when used with the `-version` flag:
```
imre@silicon:~$ cosign -version
WARNING: the flag -version is deprecated and will be removed in a future release. Please use the flag --version.
[..]
```

This is not helpful as the `--version` flag doesn't work either:
```
imre@silicon:~$ cosign --version
Error: unknown flag: --version
main.go:52: error during command execution: unknown flag: --version
```

In fact, there is no version *flag* at all. The user should be told to use the version *command* instead:
```
imre@silicon:~$ cosign version
  ______   ______        _______. __    _______ .__   __.
 /      | /  __  \      /       ||  |  /  _____||  \ |  |
|  ,----'|  |  |  |    |   (----`|  | |  |  __  |   \|  |
|  |     |  |  |  |     \   \    |  | |  | |_ | |  . `  |
|  `----.|  `--'  | .----)   |   |  | |  |__| | |  |\   |
 \______| \______/  |_______/    |__|  \______| |__| \__|
cosign: A tool for Container Signing, Verification and Storage in an OCI registry.

GitVersion:    v1.8.0
GitCommit:     9ef6b207218572b3257a5b4251418d75569baaae
GitTreeState:  clean
BuildDate:     2022-04-27T13:40:34Z
GoVersion:     go1.17.9
Compiler:      gc
Platform:      linux/amd64
```

--- a/cmd/cosign/main.go
+++ b/cmd/cosign/main.go
@@ -43,7 +43,17 @@ func main() {
 		} else if strings.HasPrefix(arg, "-") {
 			// Handle -output, convert to --output
 			newArg := fmt.Sprintf("-%s", arg)
-			fmt.Fprintf(os.Stderr, "WARNING: the flag %s is deprecated and will be removed in a future release. Please use the flag %s.\n", arg, newArg)
+			newArgType := "flag"
+			if newArg == "--version" {
+				newArg = "version"
+				newArgType = "subcommand"
+			}
+			fmt.Fprintf(
+				os.Stderr,
+				"WARNING: the %s flag is deprecated and will be removed in a future release. "+
+					"Please use the %s %s instead.\n",
+				arg, newArg, newArgType,
+			)
 			os.Args[i] = newArg
 		}
 	}

